src/segmented-control/segmented-control.scss

```diff
@@ -0,0 +1,160 @@
+    background-color: var(--color-segmented-control-button-hover-bg);
+    transition-duration: 120ms;
+  }
+
+  &:active {
+    background-color: var(--color-segmented-control-button-active-bg);
+    border-color: var(--color-segmented-control-button-active-border);
+    transition-duration: 60ms;
+  }
+
+  // Selected
+
+  &[aria-current='true'] {
+    // stylelint-disable-next-line primer/typography
+    font-weight: var(--base-text-weight-semibold, $font-weight-bold);
+    background-color: var(--color-btn-bg);
+    border-color: var(--color-segmented-control-button-selected-border);
+  }
+
+  // Divider
+
+  // stylelint-disable-next-line scss/selector-no-redundant-nesting-selector
+  & + .SegmentedControl-button::before {
+    position: absolute;
+    inset: var(--primer-borderWidth-thin, 1px) 0 0 calc(var(--primer-borderWidth-thin, 1px) * -1);
+    height: var(--primer-text-body-size-large, 16px);
+    // stylelint-disable-next-line primer/spacing
+    margin-top: var(--primer-control-medium-paddingBlock, 6px);
+    content: '';
+    // stylelint-disable-next-line primer/borders
+    border-left: var(--primer-borderWidth-thin, $border-width) $border-style var(--color-border-default);
+    transition: border-color 180ms cubic-bezier(0.3, 0.1, 0.5, 1);
+  }
+
+  // Remove dividers
+
+  &[aria-current='true'] + .SegmentedControl-button::before {
+    border-color: transparent;
+  }
+
+  &:hover,
+  &:active,
+  &:focus-visible {
+    &::before,
+    + .SegmentedControl-button::before {
+      border-color: transparent;
+      transition-duration: 60ms;
+    }
+  }
+}
+
+// Leading visual -----------------------------------------
+
+.SegmentedControl-leadingVisual {
+  color: var(--color-fg-muted);
+}
+
+// Text -----------------------------------------
+
+.SegmentedControl-text {
+  // renders a visibly hidden "copy" of the text in bold, reserving box space for when text becomes bold on selected
+  &[data-content]::before {
+    display: block;
+    height: 0;
+    // stylelint-disable-next-line primer/typography
+    font-weight: var(--base-text-weight-semibold, $font-weight-bold);
+    visibility: hidden;
+    content: attr(data-content);
+  }
+}
+
+// Variants -----------------------------------------
+
+// disabled
+.SegmentedControl--disabled { // TODO: Replace with aria-disabled?
+  .SegmentedControl-button {
+    color: var(--color-primer-fg-disabled);
+    pointer-events: none;
+    cursor: default;
+
+    &[aria-current='true'] {
+      border-color: var(--color-border-default);
+    }
+  }
+
+  .SegmentedControl-leadingVisual {
+    color: var(--color-primer-fg-disabled);
+  }
+}
+
+// fullWidth
+.SegmentedControl--fullWidth {
+  display: flex;
+
+  .SegmentedControl-button {
+    flex: 1;
+  }
+}
+
+// Icon only
+.SegmentedControl-button--iconOnly {
+  width: var(--primer-control-medium-size, 32px);
+  padding-right: 0;
+  padding-left: 0;
+}
+
+// Icon only when narrow
+@media (max-width: $width-md) { // TODO: use token
```

Comment:
What do you mean by "use token"? Isn't `$width-md` a token?

Reply:
Ok, I understand now.

Just a heads-up: `@media (max-width: var(--primer-viewportRange-regular))` won't work because you can't use custom properties in media queries yet. We'll either have to use a SCSS variable or a PostCSS plugin that swaps out the `var()` with a string.